utils/land-patterns.stanza

```diff
@@ -1071,7 +1091,154 @@ defn ipc-formula (L:Toleranced, ; the distance from edge-of-lead to edge-of-lead
+    val pad-sz-ns = transpose(pad-size(ipc-ns))
+    val pad-sz-ew = pad-size(ipc-ew)
+
+    ;----------------------------------------------------------
+    ; compute pad locations 
+    val x-shift = 0.5 * (Zmax(ipc-ew) - x(pad-sz-ew))
+    val y-shift = 0.5 * (Zmax(ipc-ns) - y(pad-sz-ns))
+
+    val pad-1-loc = to-tuple(grid-locs(num-pins-east, 2, 2.0 * x-shift, pitch-east))[0]
+    var pin-count = 0 ; accumulator, holds the number of pins we have laid out
+
+    defn layout-pads (dir:Dir, num-pins:Int, pitch:Double):
+      val locs = switch(dir):
+        Up: 
+          shift-locs{0.0, y-shift, _ } $
+          reverse(to-list(row-locs(num-pins, pitch)))
+        Down: 
+          shift-locs{0.0, (- y-shift), _ } $
+          row-locs(num-pins, pitch)
+        Left:  
+          shift-locs{(- x-shift), 0.0, _ } $
+          col-locs(num-pins, pitch)
+        Right: 
+          shift-locs{x-shift, 0.0, _ } $
+          reverse(to-list(col-locs(num-pins, pitch)))
+      
+      val pad-dims = 
+        if contains?([Left, Right], dir): pad-sz-ew
+        else: pad-sz-ns        
+      val idx = pin-count to (pin-count + num-pins)
+      for (i in idx, loc in locs) do:
+        pad p[i + 1]: smd-pad(pad-dims) at loc
+      pin-count = pin-count + num-pins
+
+    layout-pads(Left, num-pins-east,  pitch-east)
+    layout-pads(Down, num-pins-south, pitch-south)
+    layout-pads(Right,num-pins-west,  pitch-west)
+    layout-pads(Up,   num-pins-north, pitch-north)
+    
+    ;----------------------------------------------------------
+    ; Compute the courtyard
+    val part-sz = Dims(max-value(package-length), max-value(package-width))
+    val lp-sz   = enlarge(Dims(Zmax(ipc-ew), Zmax(ipc-ns)), 2.0 * SOLDER-MASK-REGISTRATION)
+    val cy-sz   = enlarge(max(part-sz, lp-sz), courtyard-excess(lead-fillets))
+    layer(Courtyard(Top)) = Rectangle(cy-sz)
+  
+    ;----------------------------------------------------------
+    ; Place the polarity marker
+    val line-width  = clearance(current-rules(), MinSilkscreenWidth)
+    val line-length = 3.0 * line-width - line-width
+    val dist  = SOLDER-MASK-REGISTRATION + SILK-SOLDER-MASK-CLEARANCE + line-width * 0.5 + 0.01
+    
+    val pad-loc = center(pad-1-loc)
+    val p1 = Point(
+      x(pad-loc) - 0.5 * x(pad-sz-ew),
+      y(pad-loc) + 0.5 * y(pad-sz-ew) + dist
+    )
+    val p2 = Point(
+      -0.5 * x(part-sz) - line-width * 0.5,
+       0.5 * y(part-sz) - line-width * 0.5,
+    )
+    val pol = max-distance(p1, p2)
+    layer(Silkscreen("pol", Top)) = Line(line-width, [
+      pol, pol + Point(line-length, 0.0)
+    ])
+
+    ;----------------------------------------------------------
+    ; Place the reference label
+    val text-y = 0.5 * (y(cy-sz) + MIN-SILKSCREEN-TEXT-HEIGHT)
+    ref-label(0.0, text-y)
+
+public defn make-quad-landpattern (num-pins:Int, 
+                                   pitch:Double, 
+                                   component-size:Toleranced, 
```

Comment:
What does component size mean? Not a standard term I'm familiar with. Is it leadspan for QFPs?

Reply:
This is repeated from the SO generator. What is "leadspan?"

Reply:
Then SO generator needs to be updated. I think `component-size` means leadspan, but use this figure to double check: https://www.pcb-3d.com/models/qfp80p900x900x160-32n_jedec_ms-026bba/

Reply:
done